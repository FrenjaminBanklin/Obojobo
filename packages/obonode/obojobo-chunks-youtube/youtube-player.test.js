import React from 'react'
import { mount, shallow } from 'enzyme'

import YouTubePlayer from './youtube-player'

jest.mock('obojobo-document-engine/src/scripts/common/util/uuid', () => {
	return () => 'mockId'
})

describe('YouTubeProperties component renders correctly', () => {
	beforeEach(() => {
		jest.clearAllMocks()
	})

	test('YouTubePlayer renders correctly', () => {
		const component = mount(<YouTubePlayer />)
		const tree = component.html()

		expect(tree).toMatchSnapshot()
	})

	test('YoutubePlayer updates video on content change', () => {
		// mock YouTube's Iframe Player object
		global.YT = {
			Player: jest.fn(() => ({
				destroy: jest.fn()
			}))
		}

		const mockContent = {
			videoId: 'mockId',
			startTime: 1,
			endTime: 2
		}
		const spy = jest.spyOn(YouTubePlayer.prototype, 'loadVideo')
		const component = shallow(<YouTubePlayer content={mockContent} />)

		expect(spy).toHaveBeenCalledTimes(1)

		component.setProps({
			content: {
				...mockContent,
				videoId: 'foo'
			}
		})

		expect(spy).toHaveBeenCalledTimes(2)

		component.setProps({
			content: {
				...mockContent,
				startTime: 4
			}
		})

		expect(spy).toHaveBeenCalledTimes(3)

		component.setProps({
			content: {
				...mockContent,
				endTime: 14
			}
		})

		expect(spy).toHaveBeenCalledTimes(4)
	})

	test("YouTubePlayer doesn't update if content hasn't changed", () => {
		global.YT = {
			Player: jest.fn(() => ({
				destroy: jest.fn()
			}))
		}

		const mockContent = {
			videoId: 'mockId',
			startTime: 1,
			endTime: 2
		}
		const spy = jest.spyOn(YouTubePlayer.prototype, 'loadVideo')
		const component = shallow(<YouTubePlayer content={mockContent} />)

		expect(spy).toHaveBeenCalledTimes(1)

		component.setProps({
			content: {
				...mockContent
			}
		})

		expect(spy).toHaveBeenCalledTimes(1)
		expect(component.instance().player.destroy).not.toHaveBeenCalled()
	})
})
