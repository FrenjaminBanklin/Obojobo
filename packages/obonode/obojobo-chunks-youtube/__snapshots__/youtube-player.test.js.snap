// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YouTubeProperties component renders correctly YouTubePlayer renders correctly 1`] = `"<div><div id=\\"obojobo-draft--chunks-you-tube-player-mockId\\" class=\\"obojobo-draft--chunks-you-tube-player\\"></div></div>"`;
